src/tests/Common/CLRTest.Jit.targets

```diff
@@ -365,6 +372,139 @@ set DOTNET_JitPath=%CORE_ROOT%\superpmi-shim-collector.dll
     </PropertyGroup>
   </Target>
 
+  <!--
+  ***********************************************************************************************
+  IlasmRoundTrip for merged tests
+  ***********************************************************************************************
+  -->
+
+  <Target Name="IlasmRoundTripCommand" AfterTargets="AfterBuild" Condition="'$(IsMergedTestRunnerAssembly)' == 'true'">
+    <PropertyGroup>
+      <InputAssemblyName>$(AssemblyName).dll</InputAssemblyName>
+      <DisassemblyName>IL-RT/$(AssemblyName).il</DisassemblyName>
+      <TargetAssemblyName>IL-RT/$(AssemblyName).dll</TargetAssemblyName>
+      <_IlasmRoundTripScriptText><![CDATA[
+import os
+import shutil
+import subprocess
+import sys
+import glob
+import struct
+import platform
+
+def is_managed_assembly(file):
+    file_size = os.path.getsize(file)
+
+    if file_size < 64:
+        return False
+
+    with open(file, "rb") as f:
+        f.seek(0x3c)
+
+        # PE Header starts @ 0x3C (60). Its a 4 byte header.
+        peHeaderPointer = struct.unpack('I', f.read(4))[0]
```

Comment:
I think you need to specify byte order for all of these.  I think that's an indication that the C# link in my other comment is a better way to go.

Reply:
This should be in the right bye order already; I've already checked the values that come out of it.

Reply:
Per https://wiki.osdev.org/PE#Overview, fields in PE files are little-endian.  Per https://docs.python.org/3/library/struct.html, `'I'` is the same as `'@I'`, which is "native" byte order.  You are relying on the machine being little-endian (which in practice today is accurate).  It looks like the size is also machine dependent.  If you switch to `'<I'`, then you get fixed little-endian, 4 byte size.

Reply:
I removed the python code and just wrote a C# tool to do the checks. We needed to check to see if the assembly was built in DEBUG anyway so we can use the proper flags for ilasm.